### PR TITLE
[OCM-16] Add RootVolume to MachineType

### PR DESCRIPTION
**Pull request by leseb**
This will allow us to set a Size and/or IOPS for a given root volume.

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
@ciaranRoche @vkareh as discussed, PTAL.

File: model/clusters_mgmt/v1/machine_type_type.model [hunk 1]
```diff
@@ -42,4 +42,7 @@ class MachineType {
 
 	// 'true' if the instance type is supported only for CCS clusters, 'false' otherwise.
 	CCSOnly Boolean
+
+	// The machine root volume capabilities.
+	RootVolume AWSVolume
```
Comment (ciaranRoche) [inline model/clusters_mgmt/v1/machine_type_type.model:47]: Is this the correct option we want to go with? What about other providers? 

Would it make sense to have a RootVolume struct and nest under that struct the provider specific volumes. 

```
type RootVolume struct {
    AWS AWSVolume
    GCP GCPVolume
    ....
}
```

What do you think @vkareh @etabak

Comment (tzvatot) [reply model/clusters_mgmt/v1/machine_type_type.model:47]: SGTM

**Review by tzvatot — CHANGES_REQUESTED** [review]

File: model/clusters_mgmt/v1/machine_type_root_volume_type.model [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Machine type root volume.
+struct MachineTypeRootVolume {
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/machine_type_root_volume_type.model:18]: Do we really need the "MachineType" prefix? @zgalor @tbrisker @vkareh WDYT?

Comment (zgalor) [reply model/clusters_mgmt/v1/machine_type_root_volume_type.model:18]: The prefix is fine

**Comment by leseb** [comment]
@gdbranco PTAL.

**Comment by tzvatot** [comment]
LGTM

**Comment by leseb** [comment]
@tzvatot thanks for the merge, what's the procedure to get a new release out?

**Comment by tzvatot** [comment]
> @tzvatot thanks for the merge, what's the procedure to get a new release out?

https://github.com/openshift-online/ocm-sdk-go/blob/main/CONTRIBUTING.md#releasing-a-new-ocm-api-model-version

**Pull request merged**